README.md

```diff
@@ -1,8 +1,67 @@
+# JSON Compilation Database Format Specification
+
+This document describes a format for specifying how to replay single link commands independently of the build system.
+
+## Background
+
+Like compilation commands, link commands are needed for code analysis, for example, look into simple C/C++ code:  
+`a.cpp:`
+```
+int f(int a) {
+  return g(a);
+}
+```
+`x.cpp:`
+```
+int g(int a) {
+  return a + 1;
+}
+```
+`y.cpp:`
+```
+int g(int a) {
+  return a - 1;
+}
 ```
-python3 main.py --i="path-to-project"
-``` 
+A tool may need to know which function is actually called, but this information is only available at linking step as soon as there may be several functions with same name in different units.  
+More example:
 
-for information use ```python3 main.py -h```
- 
+1) Unit testing tools which use code analysis need to understand which function is called in order to get correct test result.
+2) IDEs need it to understand which file to go when you click "Go to definition" on a function call.
+
+In addition, it is possible to extract link commands from build systems without executing the program.
```

Comment:
That's true, but why do you mention it? It would be weird if you needed to execute the program in order to get link commands